pkg/collector/corechecks/cluster/orchestrator.go

```diff
@@ -0,0 +1,415 @@
+	apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
+	defer apiCancel()
+	apiCl, err := apiserver.WaitForAPIClient(apiCtx)
+	if err != nil {
+		return err
+	}
+
+	// Prepare the collectors for the resources specified in the configuration file.
+	collectors := o.instance.Collectors
+
+	// Enable the KSM default collectors if the config collectors list is empty.
+	if len(collectors) == 0 {
+		collectors = defaultResources
+	}
+
+	informersToSync := map[apiserver.InformerName]cache.SharedInformer{}
+
+	for _, v := range collectors {
+		switch v {
+		case "pods":
+			podInformer := apiCl.UnassignedPodInformerFactory.Core().V1().Pods()
+			o.unassignedPodLister = podInformer.Lister()
+			o.unassignedPodListerSync = podInformer.Informer().HasSynced
+			informersToSync[apiserver.PodsInformer] = podInformer.Informer()
+		case "deployments":
+			deployInformer := apiCl.InformerFactory.Apps().V1().Deployments()
+			o.deployLister = deployInformer.Lister()
+			o.deployListerSync = deployInformer.Informer().HasSynced
+			informersToSync[apiserver.DeploysInformer] = deployInformer.Informer()
+		case "replicasets":
+			rsInformer := apiCl.InformerFactory.Apps().V1().ReplicaSets()
+			o.rsLister = rsInformer.Lister()
+			o.rsListerSync = rsInformer.Informer().HasSynced
+			informersToSync[apiserver.ReplicaSetsInformer] = rsInformer.Informer()
+		case "services":
+			serviceInformer := apiCl.InformerFactory.Core().V1().Services()
+			o.serviceLister = serviceInformer.Lister()
+			o.serviceListerSync = serviceInformer.Informer().HasSynced
+			informersToSync[apiserver.ServicesInformer] = serviceInformer.Informer()
+		case "nodes":
+			nodesInformer := apiCl.InformerFactory.Core().V1().Nodes()
+			o.nodesLister = nodesInformer.Lister()
+			o.nodesListerSync = nodesInformer.Informer().HasSynced
+			informersToSync[apiserver.NodesInformer] = nodesInformer.Informer()
+		default:
+			o.Warnf("Unsupported collector: %s", v) //nolint:errcheck
+		}
+	}
+
+	apiCl.UnassignedPodInformerFactory.Start(o.stopCh)
+	apiCl.InformerFactory.Start(o.stopCh)
+
+	return apiserver.SyncInformers(informersToSync)
+}
+
+// Run runs the orchestrator check
+func (o *OrchestratorCheck) Run() error {
+	// access serializer
+	sender, err := aggregator.GetSender(o.ID())
+	if err != nil {
+		return err
+	}
+
+	// If the check is configured as a cluster check, the cluster check worker needs to skip the leader election section.
+	// we also do a safety check for dedicated runners to avoid trying the leader election
+	if !o.isCLCRunner || !o.instance.LeaderSkip {
+		// Only run if Leader Election is enabled.
+		if !config.Datadog.GetBool("leader_election") {
+			return log.Error("Leader Election not enabled. The cluster-agent will not run the check.")
+		}
+		errLeader := o.runLeaderElection()
+		if errLeader != nil {
+			if errLeader == apiserver.ErrNotLeader {
+				// Only the leader can instantiate the apiserver client.
```

Comment:
this is a bit misleading as the apiserver client is instantiated in the Configure method